### ISSUE TITLE
combineTemplate does not use actual object instances if given directly (loses protype functions)

### ISSUE THREAD

**Issue by ottopoika** [baconjs/bacon.js#615]
```javascript
var Foo = function (){};
Foo.prototype.do = function(){console.log('FooDo');}

Bacon
  .combineTemplate({foo1: new Foo(), foo2: Bacon.once(new Foo())})
  .onValue(function(combined){
    console.log('both should return function =>', typeof combined.foo1.do, typeof combined.foo2.do)
    console.log('both should return true =>', combined.foo1 instanceof Foo, combined.foo2 instanceof Foo)
  });
```
Found out this after trying to update from 0.7.14 to 0.7.65. The older version did not use the actual instances either but at least seemed to copy inherited functions.

**Comment by raimohanska** [comment]
Ah-ha. Maybe it indeed should use the original objects if there's nothing to be replaced inside the object instance. As you've certainly figured out, you can always wrap them into Bacon.once() or Bacon.constant() as a workaround.

Would you care to submit a PR with a test and a fix?

**Comment by raimohanska** [comment]
I extracted `combineTemplate` into a separate file in 185a469. It can now be tested individually by running `./test combinetemplate`.

**Comment by ottopoika** [comment]
ok, will check if I can fix this.

**Comment by raimohanska** [comment]
I wrote a new condition for "whether to drill into an object" in commit bd9c4b4.

I included the new tests added in #615 into master.

**Comment by raimohanska** [comment]
The idea is now that templating is applied only to objects that have either `Array` or `Object` as their prototype. All tests pass.
